src/ui/ui_settings/ui_settings_service_for_request.js

```diff
@@ -0,0 +1,20 @@
+import { uiSettingsServiceFactory } from './ui_settings_service_factory';
+
+const BY_REQUEST_CACHE = new WeakMap();
+
+export function getUiSettingsServiceForRequest(server, request, readInterceptor) {
+  if (BY_REQUEST_CACHE.has(request)) {
```

Comment:
🤷‍♂️ I personally like that using `.has()` prevents the check from needing to know what a "valid" cache looks like, and just determines the existence of the cache at all, meaning that the similar caching logic will look exactly the same even if the cache contents can be "falsy" or even `null`/`undefined`. I also can't imagine that there is any perceptible performance difference